src/gleam/float.gleam

```diff
@@ -384,3 +384,17 @@ if javascript {
   external fn do_random_uniform() -> Float =
     "../gleam_stdlib.mjs" "random_uniform"
 }
+
+pub type DivisionByZero {
+  DivisionByZero
+}
+
+pub fn divide(
+  dividend dividend: Float,
+  divisor divisor: Float,
+) -> Result(Float, DivisionByZero) {
```

Comment:
This error doesn't convey anything, so let's just use Nil as we do elsewhere in the standard library.

Reply:
Just followed the existing example in the same module:

```
/// Error value when trying to operate with a base out of the allowed range.
pub type InvalidBase {
  InvalidBase
}
```